lib/offsite_payments/integrations/paytm.rb

```diff
@@ -0,0 +1,298 @@
+module OffsitePayments #:nodoc:
+  module Integrations #:nodoc:
+    module Paytm
+      require 'openssl'
+      require 'base64'
+      require 'digest'
+      require 'securerandom'
+
+      $DEBUG = false
+      #$RUN_UNIT_TEST = 0 #do not make this true, in normal
+
+      mattr_accessor :test_url
+      mattr_accessor :production_url
+
+      self.test_url = 'https://pguat.paytm.com/oltp-web/processTransaction'
+      self.production_url = 'https://secure.paytm.in/oltp-web/processTransaction'
+
+      def self.service_url
+        # self.production_url
+        OffsitePayments.mode == :production ? production_url : test_url
+      end
+
+      def self.notification(post, options = {})
+        Notification.new(post, options)
+      end
+
+      def self.return(post, options = {})
+        Return.new(post, options)
+      end
+
+      def self.checksum(chkParams, secret_key, salt = nil)
+        if salt.nil?
+          o = [('a'..'z'), ('A'..'Z'), ('0'..'9')].map { |i| i.to_a }.flatten
+          salt = (0...4).map { o[Random.rand(o.length)] }.join
```

Comment:
this logic is not working it will generate a 4 digit salt of the similar characters like "aaaa" instead of 4 digit random characters "0M3b"